nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/users/nf-users-table.js

```diff
@@ -470,6 +470,214 @@ nf.UsersTable = (function () {
+                buttonText: 'Close',
+                color: {
+                    base: '#728E9B',
+                    hover: '#004849',
+                    text: '#ffffff'
+                },
+                handler: {
+                    click: function () {
+                        //TODO: some work
+
+                        $('#user-policies-dialog').modal('hide');
+                    }
+                }
+            }]
+        });
+    };
+
+    /**
+     * Generates a human readable global policy strung.
+     *
+     * @param dataContext
+     * @returns {string}
+     */
+    var globalResourceParser = function (dataContext) {
+        return 'Global policy to ' +
+            nf.Common.getPolicyTypeListing(nf.Common.substringAfterFirst(dataContext.component.resource, '/')).text;
+    };
+
+    /**
+     * Generates a human readable component policy string.
+     *
+     * @param dataContext
+     * @returns {string}
+     */
+    var componentResourceParser = function (dataContext) {
+        var resource = dataContext.component.resource;
+        var policyLabel = '';
+
+        //determine policy type
+        if (resource.startsWith('/policies')) {
+            resource = nf.Common.substringAfterFirst(resource, '/policies');
+            policyLabel += 'Admin policy for ';
+        } else if (resource.startsWith('/data-transfer')) {
+            resource = nf.Common.substringAfterFirst(resource, '/data-transfer');
+            policyLabel += 'Site to site policy for ';
+        } else if (resource.startsWith('/data')) {
+            resource = nf.Common.substringAfterFirst(resource, '/data');
+            policyLabel += 'Data policy for ';
+        } else {
+            policyLabel += 'Component policy for ';
+        }
+
+        if (resource.startsWith('/processors')) {
+            policyLabel += 'processor ';
+        } else if (resource.startsWith('/controller-services')) {
+            policyLabel += 'controller services ';
+        } else if (resource.startsWith('/funnels')) {
+            policyLabel += 'funnel ';
+        } else if (resource.startsWith('/input-ports')) {
+            policyLabel += 'input port ';
+        } else if (resource.startsWith('/labels')) {
+            policyLabel += 'label ';
+        } else if (resource.startsWith('/output-ports')) {
+            policyLabel += 'output port ';
+        } else if (resource.startsWith('/process-groups')) {
+            policyLabel += 'process group ';
+        } else if (resource.startsWith('/remote-process-groups')) {
+            policyLabel += 'remote process group ';
+        } else if (resource.startsWith('/reporting-tasks')) {
+            policyLabel += 'reporting task ';
+        } else if (resource.startsWith('/templates')) {
+            policyLabel += 'template ';
+        }
+
+        if (dataContext.component.componentReference.permissions.canRead === true) {
+            policyLabel += dataContext.component.componentReference.component.name;
```

Comment:
Thoughts on emphasizing the component name? In some example it seems to blend in with the rest of the string.